bftengine/src/bcstatetransfer/BCStateTran.hpp

```diff
@@ -296,8 +296,6 @@ class BCStateTran : public IStateTransfer {
   // or GettingMissingResPages
   ///////////////////////////////////////////////////////////////////////////
 
-  SourceSelector sourceSelector_;
```

Comment:
Why is the move?
Source selector is used in GettingMissingResPages/GettingMissingBlocks and it is already protected.
I don't see the benefit compare to the new place it was moved to.
Also, I think It is better (unless something really needed) to add refactoring in a seperate commit or even seeperate PR. This move is unrelated to the PR and in the long term it makes much harder to dig into the history of changes when changes of different topics are mixed. (this is a general comment about the refactoring part in this PR). At least a sepearate commit makes sense here.